Client/Frontend/Browser/BrowserLocationView.swift

```diff
@@ -132,13 +138,23 @@ class BrowserLocationView : UIView, UIGestureRecognizerDelegate {
     var url: NSURL? {
         didSet {
             lockImageView.hidden = (url?.scheme != "https")
-            if let url = url?.absoluteString {
-                if url.hasPrefix("http://") ?? false {
-                    locationLabel.text = url.substringFromIndex(advance(url.startIndex, 7))
-                } else if url.hasPrefix("https://") ?? false {
-                    locationLabel.text = url.substringFromIndex(advance(url.startIndex, 8))
+
+            if let urlString = url?.absoluteString {
+
+                var truncatedURL: String
+                if let scheme = url?.scheme {
+                    truncatedURL = urlString.substringFromIndex(advance(urlString.startIndex, count(scheme) + count("://")))
```

Comment:
Desktop (and Android's) behavior is a bit nuts, but the point of this is to not highlight ANY subdomains (since they allow for easy spoofing i.e. www.google.com.realsite.com). There's a huge list of tld's here that desktop and Android use for that: https://publicsuffix.org/list/effective_tld_names.dat